### PR TITLE
Use AWS SDK v2

### PR DESCRIPTION
**Pull request by mt-clearhaus**
I have tested the changes to `lock.rb` on the PCI testing environment, but I have not tested the changes to `spec_helper.rb` and `dynamodb-mutex.gemspec`.

Currently, on PCI testing, `stanchion` is able to reserve STANs using the v2 `lock.rb`:

```
mt@ip-10-0-235-232:~$ docker exec -it compose_authy_1 bash
# export $(cat authorizer.env | grep -v -E '#' | xargs)
# irb
irb(main):001:0> require './vendor/gems/stanchion/lib/stanchion.rb'
=> true
irb(main):002:0> Stanchion.next_stan
W, [2015-12-11T09:43:03.164761 #656]  WARN -- : 3845bbc29cdd-656-4109040 No STANs available.
I, [2015-12-11T09:43:03.164840 #656]  INFO -- : 3845bbc29cdd-656-4109040 Acquired adding_mutex.
I, [2015-12-11T09:43:03.164916 #656]  INFO -- : 3845bbc29cdd-656-4109040 checking if stan.lock is stale
I, [2015-12-11T09:43:03.353322 #656]  INFO -- : 3845bbc29cdd-656-4109040 acquired stan.lock
I, [2015-12-11T09:43:03.506836 #656]  INFO -- : 3845bbc29cdd-656-4109040 reserved range 217401..217500.
I, [2015-12-11T09:43:03.533526 #656]  INFO -- : 3845bbc29cdd-656-4109040 released lock stan.lock
I, [2015-12-11T09:43:03.534633 #656]  INFO -- : 3845bbc29cdd-656-4109040 Got new range: 217401..217500
I, [2015-12-11T09:43:03.534858 #656]  INFO -- : 3845bbc29cdd-656-4109040 Acquired stans_mutex.
I, [2015-12-11T09:43:03.535093 #656]  INFO -- : 3845bbc29cdd-656-4109040 Adding 217401..217500 to reserved STANs.
I, [2015-12-11T09:43:03.535510 #656]  INFO -- : 3845bbc29cdd-656-4109040 Reserved STANs: [217401, ..., 217500].
I, [2015-12-11T09:43:03.535868 #656]  INFO -- : 3845bbc29cdd-656-4109040 Releasing stans_mutex.
I, [2015-12-11T09:43:03.536080 #656]  INFO -- : 3845bbc29cdd-656-4109040 Releasing adding_mutex.
I, [2015-12-11T09:43:03.536488 #656]  INFO -- : 3845bbc29cdd-656-4109040 STAN=217401
=> 217401
```

I have not stressed the code in any way in order to test the mutual exclusion, but the idea of using `put_item` with `:unless_exists => :id` (now `condition_expression: 'attribute_not_exists(id)'`) is not changed. I have tested that a stale lock is released. I like `created` to be a string, rather than some sort of float, e.g. 0.1449827417E10, but changing this ruins backwards compatibility and might not be acceptable.

In essence, the code needs some love before being trusted in any way.

@jsmpereira @ct-clearhaus :eyes:

### CODE REVIEW

File: lib/dynamodb_mutex/lock.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
-require 'aws-sdk-v1'
+require 'aws-sdk'
+require 'socket'
```
Comment (ct-clearhaus) [inline lib/dynamodb_mutex/lock.rb]: Is `socket` needed?

Comment (mt-clearhaus) [inline lib/dynamodb_mutex/lock.rb]: It is needed for `Socket.gethostname` in the `pid` method. I think `aws-sdk-v1` loaded it previously, as the `require` became necessary when `aws-sdk-v1` was replaced with `aws-sdk`.

**Comment by mt-clearhaus** [comment]
As mentioned in the comment https://github.com/clearhaus/stanchion/pull/5#issuecomment-164386066 , this PR is a precursor for the PR https://github.com/clearhaus/stanchion/pull/5 .

@jsmpereira :eyes: when you have a moment, but I understand that it is not simple and that it is not high priority. I think @ct-clearhaus already reviewed briefly.

File: lib/dynamodb_mutex/lock.rb (hunk 1 above)
Comment (ct-clearhaus) [inline lib/dynamodb_mutex/lock.rb]: :+1:

**Comment by mt-clearhaus** [comment]
I am aware that the Travis CI build fails. I have looked at the traces without success :-(

**Comment by jsmpereira** [comment]
@mt-clearhaus I'm taking a quick look :stuck_out_tongue:

**Comment by mt-clearhaus** [comment]
@jsmpereira Thank you very much for the icing on the cake :-D

How about a version bump to 0.0.4?

I think we "just" need to convince ourselves that a) error checking in the `dynamodb_client` method is sufficient and correct, and b) mutual exclusion is implemented correctly in the `create` method.

**Comment by mt-clearhaus** [comment]
Maybe we should also poke Dinesh, such that he may update the gem on rubygems.org.

**Comment by jsmpereira** [comment]
@mt-clearhaus I can publish to rubygems.org when merged.

**Comment by jsmpereira** [comment]
I say :+1: to your stuff :stuck_out_tongue:

**Pull request merged**